chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -95,6 +106,33 @@ private[chisel3] trait HasId {
   private[chisel3] def setRef(parent: HasId, index: Int): Unit = setRef(Index(Node(parent), ILit(index)))
   private[chisel3] def setRef(parent: HasId, index: UInt): Unit = setRef(Index(Node(parent), index.ref))
   private[chisel3] def getRef: Arg = _ref.get
+
+  // Implementation of public methods.
+  def signalName = {
+    def constructName(arg: Arg): String = arg match {
+      case Node(id) => id.signalName
+      case ILit(n) => n.toString
+      case ULit(n, w) => n.toString
+      case SLit(n, w) => n.toString
+      case Ref(name) => name
+      case Slot(imm, name) => s"${imm.id.signalName}.$name"
+      case Index(imm, value) => s"${constructName(imm)}[${constructName(value)}]"
+      case ModuleIO(mod, name) => name
+    }
```

Comment:
That `fullName` requires a `Component` as its argument, so I didn't want to handle it in `Module` or somewhere else. Probably I can use it instead here.